smart-hadoop-support/smart-hadoop/src/main/java/org/smartdata/hdfs/action/CopyFileAction.java

```diff
@@ -100,41 +102,47 @@ public void init(Map<String, String> args) {
     if (args.containsKey(LENGTH)) {
       length = Long.parseLong(args.get(LENGTH));
     }
+    if (args.containsKey(COPY_CONTENT)) {
+      copyContent = Boolean.parseBoolean(args.get(COPY_CONTENT));
+    }
   }
 
   @Override
   protected void execute() throws Exception {
-    if (srcPath == null) {
-      throw new IllegalArgumentException("File parameter is missing.");
-    }
-    if (destPath == null) {
-      throw new IllegalArgumentException("Dest File parameter is missing.");
-    }
+    validateArgs();
     preserveAttributes = parsePreserveAttributes();
+
     appendLog(
-        String.format("Action starts at %s : Read %s",
-            Utils.getFormatedCurrentTime(), srcPath));
-    if (!dfsClient.exists(srcPath)) {
-      throw new ActionException("CopyFile Action fails, file doesn't exist!");
-    }
-    appendLog(
-        String.format("Copy from %s to %s", srcPath, destPath));
+        String.format("Action starts at %s : Copy from %s to %s",
+            Utils.getFormatedCurrentTime(), srcPath, destPath));
 
     srcFileStatus = getFileStatus(srcPath);
 
-    if (offset == 0 && length == 0) {
+    if (!copyContent) {
+      appendLog("Src and dest files are equal, no need to copy content");
+    } else if (offset == 0 && length == 0) {
       copySingleFile(srcPath, destPath);
-    }
-    if (length != 0) {
+    } else if (length != 0) {
       copyWithOffset(srcPath, destPath, bufferSize, offset, length);
     }
 
-    // we already preserved replication number
-    preserveAttributes.remove(REPLICATION_NUMBER);
     copyFileAttributes(srcPath, destPath, preserveAttributes);
+
     appendLog("Copy Successfully!!");
   }
 
+  private void validateArgs() throws Exception {
+    if (srcPath == null) {
```

Comment:
Should we check this field not only for null but also ensure it is not empty?

Reply:
Fixed